doc/help/elektra-data-structures.md

```diff
@@ -366,3 +366,46 @@ The algorithm uses the `syncbits` to decide if the key set needs to be
 synchronised.
 
 Continue reading [with the error handling](elektra-error-handling.md).
+
+## Order Preserving Minimal Perfect Hash Map
+
+All structs are defined [in](/src/include/opmphm.h).
+
+### Vstack
+
+The `Vstack` is a virtual stack implementation with dynamical memory allocation.
+The allocation doubles the space if the `Vstack` is full and reduces the space by the half
+if the memory usage gets under a quarter.
+
+The data will be stored as void pointer and can be inserted with the push call.
+Retrieve is possible through the pop call.
+Besides the classical init and del functions, an is-empty check is also implemented.
+
+The following [link](/src/libs/elektra/opmphm_vstack.c) leads to the code and
+[this](@ref Vstack) one to the docu.
+
+### Vheap
+
+The `Vheap` is a virtual heap implementation with dynamical memory allocation.
+A heap is an efficient data structure, if an ordered retrieval is needed.
+The allocation works like the `Vstack` allocation.
+
+The data will be stored as void pointer and can be inserted with the insert call.
+At the insert the data will be stored in a binary tree, where each parent and his childs obey the order.
```

Comment:
> At the **insert** the data will be stored in a binary tree, where each parent and his **childs** obey the order.

Maybe better: At **the point of insertion** the data will be stored in a binary tree, where each parent and his **children** obey the order.